### PR TITLE
I1992 - return populated rows for templates

### PR DESCRIPTION
**Pull request by littledice**

### CODE REVIEW

File: src/main/kotlin/org/vaccineimpact/api/models/Expectations.kt [hunk 1]
```diff
@@ -14,31 +17,70 @@ data class Expectations(
...
+    fun expectedStochasticRows(disease: String): Sequence<ExpectedStochasticRow> = buildSequence {
+        for (age in ages)
+        {
+            for (country in countries)
+            {
+                yieldAll(years
+                        .filter { (it - age).withinCohortRange() }
+                        .map { mapStochasticRow(disease, it, age, country) })
+            }
+        }
```
Comment (MartinEden) [inline src/main/kotlin/org/vaccineimpact/api/models/Expectations.kt]: Feels like more of the code could be shared between `expectedStochasticRows` and `expectedCentralRows`?

Comment (littledice) [reply src/main/kotlin/org/vaccineimpact/api/models/Expectations.kt]: I played around with that, but couldn't find a nice way to both share the code, and keep the return types distinguished. In the end I thought a bit of repetitive logic was worth it for clearly distinguished methods with their own type signatures.

Comment (MartinEden) [reply src/main/kotlin/org/vaccineimpact/api/models/Expectations.kt]: How about:

```
data class RowDeterminer(year: Int, age: Int, country: Country)

fun expectedCentralRows(diease: String) = expectedRows()
	.map { mapCentralRow(disease, it.year, it.age, it.country) }

fun expectedStochasticRows(diease: String) = expectedRows()
	.map { mapStochasticRow(disease, it.year, it.age, it.country) }

private fun expectedRows(): Sequence<RowDeterminer> = buildSequence {
    for (age in ages)
    {
        for (country in countries)
        {
        	yieldAll(years
        		     .filter { (it - age).withinCohortRange() }
                     .map { RowDeterminer(it, age, country) })

        }
    }
}
```
fun expectedCentralRows(diease: String) = expectedRows()
	.map { mapCentralRow(disease, it.year, it.age, it.country) }

fun expectedStochasticRows(diease: String) = expectedRows()
	.map { mapStochasticRow(disease, it.year, it.age, it.country) }
```

Comment (littledice) [reply src/main/kotlin/org/vaccineimpact/api/models/Expectations.kt]: That works, thanks :)

**Pull request merged**